deps/wazuh_testing/wazuh_testing/authd.py

```diff
@@ -0,0 +1,197 @@
+
+
+def validate_authd_response(response, expected):
+    """
+    Validates if the different items of an Authd response are as expected. Any item inexistent in expected won't
+    be validated.
+
+    Args:
+        response (str): The Authd response to be validated.
+        expected (dict): Dictionary with the items to validate.
+    """
+    response = response.split(sep=" ", maxsplit=1)
+    status = response[0]
+    result = 'success'
+    err_msg = ''
+    if expected['status'] == 'success':
+        result, err_msg = validate_argument(status, 'OSSEC', 'status')
+        if result != 'success':
+            return result, err_msg
+
+        agent_key = response[1].split('\'')[1::2][0].split()
+        id = agent_key[0]
+        name = agent_key[1]
+        ip = agent_key[2]
+        key = agent_key[3]
+
+        if 'id' in expected:
+            result, err_msg = validate_argument(id, expected['id'], 'id')
+            if result != 'success':
+                return result, err_msg
+
+        if 'name' in expected:
+            result, err_msg = validate_argument(name, expected['name'], 'name')
+            if result != 'success':
+                return result, err_msg
+
+        if 'ip' in expected:
+            result, err_msg = validate_argument(ip, expected['ip'], 'ip')
+            if result != 'success':
+                return result, err_msg
+
+        if 'key' in expected:
+            result, err_msg = validate_argument(key, expected['key'], 'key')
+            if result != 'success':
+                return result, err_msg
+
+    elif expected['status'] == 'error':
+        result, err_msg = validate_argument(status, 'ERROR:', 'status')
+        if result != 'success':
+            return result, err_msg
+
+        message = response[1]
+        if 'message' in expected:
+            if re.match(expected['message'], message) is None:
+                return 'error', f"Invalid 'message': '{message}' received, '{expected['message']}' expected"
+    else:
+        raise Exception('Invalid expected status')
+
+    return result, err_msg
+
+
+def clean_agents_from_db():
+    """
+    Clean agents from DB
+    """
+    command = 'global sql DELETE FROM agent WHERE id != 0'
+    try:
+        query_wdb(command)
+    except Exception:
+        raise Exception('Unable to clean agents')
+
+
+def insert_agent_in_db(id=1, name="TestAgent", ip="any", registration_time=0, connection_status=0,
+                       disconnection_time=0):
+    """
+    Write agent in global.db
+    """
+    insert_command = f'global insert-agent {{"id":{id},"name":"{name}","ip":"{ip}","date_add":{registration_time}}}'
+    update_command = f'global sql UPDATE agent SET connection_status = "{connection_status}",\
+                       disconnection_time = "{disconnection_time}" WHERE id = {id};'
+    try:
+        query_wdb(insert_command)
+        query_wdb(update_command)
+    except Exception:
+        raise Exception(f'Unable to add agent {id}')
```

Comment:
Same that before, these functions could be useful for the rest of the modules and they are more related with `wazuh_db` module.